packages/editor-ui/src/components/mixins/workflowHelpers.ts

```diff
@@ -457,8 +457,29 @@ export const workflowHelpers = mixins(
 						workflowDataRequest.tags = tags;
 					}
 
+					// Check if trigger was removed from active workflow
+					let triggerIsRemoved = false;
+					if (this.$store.getters.isActive) {
+						const triggers = this.$store.getters.allNodes
+							.map(({ type }: INodeUi) => this.$store.getters.nodeType(type))
+							.filter(((type: INodeTypeDescription) => type.group.includes('trigger')));
+						triggerIsRemoved = triggers.length === 0;
+
+						if (triggerIsRemoved) {
+							// Only set active if trigger was removed
+							workflowDataRequest.active = false;
+						}
+					}
+
 					const workflowData = await this.restApi().updateWorkflow(currentWorkflow, workflowDataRequest);
 
+					// Explicitly commit active state change
+					if (triggerIsRemoved) {
```

Comment:
why do we need this flag? cannot we simply update in line 470?

Reply:
I believe, the commits should happen once the update request is successful on line 474.